src/vaults/yields.utils.ts

```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
     };
   });
 }
+
+export async function estimateVaultPerformance(
+  chain: Chain,
+  vault: VaultDefinitionModel,
+  data: VaultHarvestData[],
+): Promise<YieldSource[]> {
+  if (data.length === 0) {
+    return [];
+  }
+
+  const sdk = await chain.getSdk();
+  const recentHarvests = data.sort((a, b) => b.timestamp - a.timestamp);
```

Comment:
new behavior of the yield function is to always assume the period is 15 days - this makes it simpler for the function (should capture in a comment above) to just base all the yield within one window